Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -257,14 +268,81 @@ public bool TryUpdateQueryBufferForLayerMask(LayerMask layerMask, Vector3 pointe
+                    Collider collider = queryBuffer[i];
+                    grabbable = collider.GetComponent<NearInteractionGrabbable>();
+                    // Additional check: is grabbable in the camera frustrum
                     if (grabbable != null)
+                    {
+                        if (!isInFOV(collider))
+                        {
+                            grabbable = null;
+                        }
+                    }
+                    if (grabbable != null)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+
+            /// <summary>
+            /// Returns true if a collider's bounds is within the camera FOV
+            /// </summary>
+            /// <param name="myCollider">The collider to test</param>
+            /// <returns></returns>
+            private bool isInFOV(Collider myCollider)
+            {
+                Bounds b = myCollider.bounds;
+                var corners = new List<Vector3>();
+                BoundsExtensions.GetColliderBoundsPoints(myCollider, corners, 0);
+                float xMin = float.MaxValue, yMin = float.MaxValue, zMin = float.MaxValue;
+                float xMax = float.MinValue, yMax = float.MinValue, zMax = float.MinValue;
+                foreach (var pt in corners)
+                {
+                    if (isPointInFrustrum(pt))
                     {
                         return true;
                     }
+
+                    xMin = Mathf.Min(xMin, pt.x);
+                    yMin = Mathf.Min(yMin, pt.y);
+                    zMin = Mathf.Min(zMin, pt.z);
+                    xMax = Mathf.Max(xMax, pt.x);
+                    yMax = Mathf.Max(yMax, pt.y);
+                    zMax = Mathf.Max(zMax, pt.z);
+                }
+
+                // edge case: check if camera is inside the entire bounds of the collider;
+                var cameraPos = CameraCache.Main.transform.position;
+                if (xMin <= cameraPos.x && cameraPos.x <= xMax 
+                    && yMin <= cameraPos.y && cameraPos.y <= yMax
+                    && zMin <= cameraPos.z && cameraPos.z <= zMax)
+                {
+                    return true;
                 }
                 return false;
             }
+
+            /// <summary>
+            /// Returns true if a point is in the camera's frustrum
+            /// </summary>
+            /// <param name="pt">Point to test</param>
+            private bool isPointInFrustrum(Vector3 pt)
+            {
+                Camera mainCam = CameraCache.Main;
+                Vector3 screenPos = mainCam.WorldToScreenPoint(pt);
+                // pixel buffer around screen
+                float screenBuffer = 10;
+                // distance buffer in front of camera, in meters
+                float cameraBuffer = 0.05f;
```

Comment:
Similar question around camera buffer here - why do we choose .05 meters (instead of say, the camera near clip plane or something else based on what the user has configured)

Reply:
Yes you're correct -- I just forgot to clean this up.